samples/ar-view-node/src/main/java/io/github/sceneview/sample/arviewnode/utils/NodeRotationHelper.kt

```diff
@@ -0,0 +1,64 @@
+package io.github.sceneview.sample.arviewnode.utils
+
+import com.google.ar.core.TrackingState
+import io.github.sceneview.ar.ARSceneView
+import io.github.sceneview.ar.node.AnchorNode
+import io.github.sceneview.collision.Quaternion
+import io.github.sceneview.collision.Vector3
+import io.github.sceneview.math.toVector3
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+
+class NodeRotationHelper {
+
+    private var skipCounter = 0
+    private var isRunning = false
+
+    fun updateRotationOfViewNodes(
+        sceneView: ARSceneView,
+        nodeList: MutableList<AnchorNode>
+    ) {
+        if (isRunning) {
+            return
+        }
+
+        if (skipCounter <= MAX_SKIP_FRAMES) {
+            skipCounter++
+            return
+        }
+
+        skipCounter = 0
+
+        GlobalScope.launch {
+            isRunning = true
+
+            nodeList.forEach {
+                if (it.anchor.trackingState == TrackingState.TRACKING) {
+                    val cameraPosition = sceneView.cameraNode.worldPosition
+
+                    val nodePosition = it.worldPosition
+
+                    val cameraVec3 = cameraPosition.toVector3()
+                    val nodeVec3 = nodePosition.toVector3()
+
+                    val direction = Vector3.subtract(cameraVec3, nodeVec3)
+
+                    val lookRotation = Quaternion.lookRotation(direction, Vector3.up())
+
+                    it.worldQuaternion = dev.romainguy.kotlin.math.Quaternion(
+                        lookRotation.x,
+                        lookRotation.y,
+                        lookRotation.z,
+                        lookRotation.w)
+                }
+            }
+
+            isRunning = false
+        }
```

Comment:
I think this block is still executed on the main frame but, probably, at a later time.

Reply:
Ah ok, I don't use coroutines very often. I should switch the good old ExecutorService with some AtomicBooleans :O :P

Reply:
Is it enough to wrap the code into

```kotlin
withContext(Dispatchers.IO) {

}
```

?

Reply:
Sceneview should only be accessed from the main thread so I think such optimization isn't possible here. We can leave the code on the main thread if it isn't causing any performance issues.

Reply:
I kept it for now, so far I had no issues.